packages/core/app-layout/src/components/navbar/AccountDropdown.vue

```diff
@@ -0,0 +1,101 @@
+<template>
+  <KDropdownMenu
+    class="account-dropdown"
+    :kpop-attributes="{ placement: 'bottomStart' }"
+    :label="userInitials"
+    size="small"
+    :width="width"
+    @change="(selection: any) => optionSelected(selection)"
+  >
+    <template #items="{ handleSelection }">
+      <slot>
+        <KDropdownItem
+          v-for="option in options"
+          :key="option.value"
+          :class="{ 'has-divider': option.hasDivider }"
+          :data-testid="'select-' + option.value"
+          :has-divider="option.hasDivider"
+          :selected="selectedOption.value === option.value"
+          @click="handleSelection(option)"
+        >
+          <slot :option="option">
+            {{ option.label }}
+          </slot>
+        </KDropdownItem>
+      </slot>
+    </template>
+  </KDropdownMenu>
+</template>
+
+<script setup lang="ts">
+import { ref, Ref, PropType } from 'vue'
+import type { DropdownItem } from '@kong/kongponents'
+
+const props = defineProps({
+  options: {
+    type: Array as PropType<DropdownItem[]>,
+    required: true,
+  },
+  width: {
+    type: String,
+    default: '240',
+  },
+  userInitials: {
+    type: String,
+    required: true,
+  },
+})
+const emit = defineEmits(['change'])
+
+const selectedOption: Ref<DropdownItem> = ref(props.options?.filter((option: DropdownItem) => !!option.selected)[0] || props.options[0])
+
+const optionSelected = (option: DropdownItem) => {
+  selectedOption.value = option
+  emit('change', selectedOption.value)
+}
+
+</script>
+
+<style lang="scss" scoped>
+.account-dropdown {
+  display: flex;
+  margin-right: 4px;
+  :deep(.k-button.primary), :deep(.k-button.primary:hover:not(:disabled)) {
+    background-color: var(--kong-ui-account-dropdown-background, var(--purple-300, #9396FC));
+    color: var(--kong-ui-account-dropdown-color, var(--white, #fff));
+    height: 32px;
+    justify-content: center;
+    padding: 0;
+    width: 32px;
+  }
+  // Align the dropdown to the bottom of the navbar
+  :deep(.k-dropdown-popover) {
+    top: 10px !important;
+  }
+
+  :deep(.k-dropdown-item:first-child) {
+    > button {
+      color: var(--grey-600, #3C4557);
+      font-weight:600 !important;
+      padding-bottom: 0px;
+    }
+  }
+  :deep(.k-dropdown-item:nth-child(2)) {
+    > button {
+      color: var(--grey-500, #6F7787);
+      font-size:14px;
+      padding-top: 4px;
+    }
+  }
+
+  :deep(.k-dropdown-item.has-divider:before) {
+    background-color: var(--grey-300, #E7E7EC);
+  }
+
+  :deep(.k-dropdown-item:nth-child()) {
+    > button {
+      color: var(--grey-600, #3C4557);
+    }
+  }
```

Comment:
Though it is a generic component, since it is specifically an AccountDropdown, it's reasonable to assume all implementations of an AccountDropdown will have the User's Name, Email, and then the rest of the items.

Reply:
<img width="335" alt="Screenshot 2023-03-20 at 10 25 49 PM" src="https://user-images.githubusercontent.com/2568272/226683891-6ff539cf-2a8b-4f82-b9af-9bb61b3c0e86.png">

Reply:
The first element is inaccurate in the sandbox regardless; the email wouldn't be its own item, it would be slotted in with the name in the same item. Then the `nth-child` rules are off for the rest of the items, and we will have to modify them every time Product adds/removes items in the dropdown.

Reply:
> These are for applying custom styles to the [Name](https://www.figma.com/file/atUr0wwTnwFehr7LjgCQOb/My-Account?node-id=667%3A3269&t=NDHd2NR4C71Poi3V-0) (bold), [Email](https://www.figma.com/file/atUr0wwTnwFehr7LjgCQOb/My-Account?node-id=667%3A3269&t=NDHd2NR4C71Poi3V-0) (smaller font size and different color) and the [rest of the items](https://www.figma.com/file/atUr0wwTnwFehr7LjgCQOb/My-Account?node-id=667%3A3269&t=NDHd2NR4C71Poi3V-0) per the Figma Mock.

Do you think it's better to have a specific class name for those items (names, email) and target that element specifically?

Reply:
No class names, etc. should exist in this repository. The **host app** controls these items and should be responsible for styling the slotted content

Reply:
Ok, I have removed the custom styling from here, and will instead put them in the host app (khcp-ui).

Reply:
I haven't gotten the slotting of email with name working yet. I figured I could do it in khcp-ui, and unblock this pr from merging.

Reply:
> No class names, etc. should exist in this repository. The **host app** controls these items and should be responsible for styling the slotted content

ah correct! For a second I forgot it's the shared-ui component. Yes, it should be slottable and consuming app can decide how to display contents inside the dropdown and its styles.

Reply:
> I haven't gotten the slotting of email with name working yet

You should just be able to do something like this and then apply styles as needed:

```html
<KDropdownItem>
  <div>{{ name }}</div>
  <div>{{ email }}</div>
</KDropdownItem>
```

Reply:
Aah but Im not explicitly defining each KDropdownItem, instead I'm using an array `options` of objects `label: 'someLabel', to: '/'` and passing that as props to `KDropdownItem`

Reply:
I have also tried using `\n` and `<br/>` between the username and email to make them render together, but no luck.

Reply:
I think in the host app we would slot in the item instead of looping through it. We could do something like `v-if="option.label === 'something'` use a specific KDropdownItem template, otherwise just use the default.